### PR TITLE
Handle exceptions caused by partitioned tables

### PR DESCRIPTION
**Pull request by Leo-XM-Zeng**
Because we do not yet have good support for partitioned tables, the following code triggers assertions. #409

```c
/*
 * RelationGetNumberOfBlocksInFork
 *		Determines the current number of pages in the specified relation fork.
 *
 * Note that the accuracy of the result will depend on the details of the
 * relation's storage. For builtin AMs it'll be accurate, but for external AMs
 * it might not be.
 */
BlockNumber
RelationGetNumberOfBlocksInFork(Relation relation, ForkNumber forkNum)
{
	if (RELKIND_HAS_TABLE_AM(relation->rd_rel->relkind))
	{
		/*
		 * Not every table AM uses BLCKSZ wide fixed size blocks. Therefore
		 * tableam returns the size in bytes - but for the purpose of this
		 * routine, we want the number of blocks. Therefore divide, rounding
		 * up.
		 */
		uint64		szbytes;

		szbytes = table_relation_size(relation, forkNum);

		return (szbytes + (BLCKSZ - 1)) / BLCKSZ;
	}
	else if (RELKIND_HAS_STORAGE(relation->rd_rel->relkind))
	{
		return smgrnblocks(RelationGetSmgr(relation), forkNum);
	}
	else
		Assert(false);

	return 0;					/* keep compiler quiet */
}
``` 
**What does this PR do?**

When accessing the partition table, we temporarily let PG handle it instead of DuckDB.
This temporarily avoids the unavailability of the system due to an assertion.

### CODE REVIEW

**Comment by Leo-XM-Zeng** [comment]
Of course we're going to have to deal with #19 eventually.

File: src/pgduckdb_hooks.cpp [hunk 1]
```diff
@@ -52,6 +52,23 @@ IsCatalogTable(List *tables) {
 	return false;
 }
 
+static bool
+IsPartitionedTable(List *tables) {
```
Comment (JelteF) [inline src/pgduckdb_hooks.cpp:56]: I think `Is` is the wrong prefix here and the argument name `tables` is also confusing. We pass it a list of tables RTEs not a single table. I realize the problem is the case for `IsCatalogTable`, but we should change this there too (not in this PR though).
```suggestion
ContainsPartitionedTable(List *rtes) {
```

File: src/pgduckdb_hooks.cpp [hunk 2]
```diff
@@ -52,6 +52,23 @@ IsCatalogTable(List *tables) {
...
+	foreach_node(RangeTblEntry, table, tables) {
+		if (table->rtekind == RTE_SUBQUERY) {
+			/* Check whether the table in the subquery is a partitioned table */
```
Comment (JelteF) [inline src/pgduckdb_hooks.cpp:59]:
```suggestion
			/* Check whether any table in the subquery is a partitioned table */
```

**Review by JelteF — COMMENTED** [review]
Thank you for the fix. Some small naming suggestions, but other than that this looks good.

**Review by JelteF — APPROVED** [review]

File: src/pgduckdb_hooks.cpp (hunk 1 above)
Comment (Leo-XM-Zeng) [reply src/pgduckdb_hooks.cpp]: Updated the IsCatalogTable function information #417

**Pull request merged**